### PR TITLE
[WIP] Port to GTK4

### LINKED ISSUE #610: Move to GTK4 and libadwaita
GTK4 is now supported by a large number of distributions. Furthermore, libhandy (used for many widgets) has become libadwaita and it is better to migrate now before the required effort becomes (for me) unmanageable.

- https://gitlab.gnome.org/GNOME/libadwaita
- https://docs.gtk.org/gtk4/

Some distributions may not offer libadwaita out of the box (or at all) but I don't see a limitation, as we provide the flatpak, appimage and snap version for these reasons as well.

### PR DESCRIPTION
**Pull request by jannuary**
Let's get this going. Still very broken and the commits are a mess. Fixes #610, would also port things to new widgetry where possible/or needed.

Checklistt:

- [ ] **python**
  - [x] **misc**
    - [x] remove all gtk_dialog.run's
    - [x] fix InstallerManager.__ask_for_local_resources()
    - [x] running operations are hidden in the main view, no feedback is given to the user

  - [x] **dialogs**
    - [x] crash
    - [x] dlloverrides
    - [x] drives
    - [x] duplicate
    - [x] envvars
    - [x] gamescope
    - [x] generic
    - [x] generic_cli
    - [x] journal
    - [x] onboard
    - [x] rename
    - [x] runargs

  - [x] **views**
    - [x] bottle_dependencies
    - [x] bottle_details
    - [x] bottle_installers
    - [x] bottle_preferences
    - [x] bottle_programs
    - [x] bottle_taskmanager
    - [x] bottle_versioning
    - [x] details
    - [x] importer
    - [x] library
    - [x] list
    - [x] loading
    - [x] new
    - [x] preferences

  - [x] **widgets**
    - [x] component
    - [x] dependency
    - [x] executable
    - [x] importer
    - [x] installer
    - [x] library
    - [x] page
    - [x] program
    - [x] state

  - [x] cli.in (need test)
  - [x] main
  - [x] operation
  - [x] operation_cli
  - [x] window

- [x] **ui**
  - [x] about
  - [x] component-entry
  - [x] dependency-entry
  - [x] details
  - [x] details-bottle
  - [x] details-dependencies
  - [x] details-installers
  - [x] details-preferences
  - [x] details-programs
  - [x] details-versioning
  - [x] dialog-crash-report
  - [x] dialog-dll-overrides
  - [x] dialog-drives
  - [x] dialog-duplicate
  - [x] dialog-env-vars
  - [x] dialog-gamescope
  - [x] dialog-journal
  - [x] dialog-launch-options
  - [x] dialog-rename
  - [x] dialog-run-args
  - [x] dll-override-entry
  - [x] drive-entry
  - [x] env-var-entry
  - [x] importer
  - [x] importer-entry
  - [x] installer-entry
  - [x] library
  - [x] library-entry
  - [x] list
  - [x] list-entry
  - [x] loading
  - [x] new
  - [x] onboard
    - [x] replace bottles-welcome.png draft
  - [x] preferences
  - [x] program-entry
  - [x] state-entry
  - [x] task-entry
  - [x] window

- [ ] **Not blocking/Follow-up**
  - [ ] port ComboBox'es to ComboRow 
  - [x] use Toasts where applicable
  - [ ] use menu models
  - [x] implement AdwStatusPage for installers
  - [ ] implement AdwStatusPage for dependencies
  - [x] new Onboard 

- [x] **deprecations**
  - [x] message.py
  - [x] dialog.ui
  - [x] empty.ui
  - [x] message-entry.ui
  - [x] environment-variables.ui
  - [x] elementary.css
  - [x] yaru.css

<details>
  <summary>Old checklist</summary>

```
Checklist:

- [x] about.ui - needs to be a child of the main window
- [x] component-entry.ui
- [x] dependency-entry.ui
- [x] details-*.ui
- [x] ~~dialog.ui~~ deprecated
- [ ] dialog-*.ui
- [ ] drive-entry.ui 
- [x] ~~empty.ui~~ deprecated
- [x] env-var-entry.ui
- [x] importer.ui
- [x] importer-entry.ui
- [x] installer-entry.ui
- [x] library.ui
- [x] library-entry.ui
- [x] loading.ui
- [x] ~~message-entry.ui~~ deprecated
- [x] ~~new.ui~~ redone 
- [ ] onboard.ui (missing final illustration)
- [x] preferences.ui
- [x] program-entry.ui
- [x] state-entry.ui
- [x] window.ui
- [ ] remove all gtk_dialog.run's

- Not blocking:
- [ ] port ComboBox'es to ComboRow 
- [ ] use Toasts where applicable
- [ ] use menu models
```
</details>

### CODE REVIEW

File: src/dialogs/drives.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def __remove(self, widget):
 
 
 @Gtk.Template(resource_path='/com/usebottles/bottles/dialog-drives.ui')
-class DrivesDialog(Handy.Window):
+class DrivesDialog(Adw.Window):
```
Comment (jannuary) [inline src/dialogs/drives.py:81]: ideally we wouldn't have this and other as AdwWindow's - they were only Hdy* for rounded corners, which is unneccesary now

File: src/dialogs/launchoptions.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def __init__(self, parent, config, program, **kwargs):
 
     def __close_window(self, widget=None):
         self.parent.page_details.set_config(self.config)
-        self.destroy()
+        self.window.remove(self)
```
Comment (jannuary) [inline src/dialogs/launchoptions.py:63]: windows can and should destroy themselves

File: src/main.py [hunk 3]
```diff
@@ -181,18 +182,12 @@ def do_startup(self):
         Gtk.Application.do_startup(self)
         self.__register_actions()
 
-        # Opt-in to follow dark mode user preference.
-        # TODO Remove after porting to libadwaita.
-        manager = Handy.StyleManager.get_default()
-        manager.set_color_scheme(Handy.ColorScheme.PREFER_LIGHT)
-
-        # TODO AdwApplication manages this automatically for us.
         self.dark_provider = Gtk.CssProvider()
```
Comment (jannuary) [inline src/main.py:185]: do we need this?

Comment (mirkobrombin) [reply src/main.py:185]: No more needed I think

File: com.usebottles.bottles.dev.yml [hunk 4]
```diff
@@ -273,6 +258,30 @@ modules:
...
+  
+  - name: libadwaita
+    buildsystem: meson
+    config-opts:
+    - -Dexamples=false
+    - -Dtests=false
+    sources:
+    - type: git
+      url: https://gitlab.gnome.org/GNOME/libadwaita.git
+      commit: 39a9a4b5b5987737537d6a044da7c7fdb91f544f
```
Comment (jannuary) [inline com.usebottles.bottles.dev.yml:270]: This is 1.0 though?

Comment (mirkobrombin) [reply com.usebottles.bottles.dev.yml:270]: 1.0.4

Comment (mirkobrombin) [reply com.usebottles.bottles.dev.yml:270]: we can switch to `5ec9f624b23bf78b29dd708e77ccdfcee0a9867a` for 1.1.1 which should be the most recent

Comment (jannuary) [reply com.usebottles.bottles.dev.yml:270]: Either way, let's either keep the yml as is or delete it outright - we have the json manifest already

Comment (mirkobrombin) [reply com.usebottles.bottles.dev.yml:270]: I prefer and use yaml, it's also more clear to read

File: src/main.py (hunk 3 above)
Comment (jannuary) [reply src/main.py:185]: hm, it doesn't seem like the styles are automatically loaded

Comment (jannuary) [reply src/main.py:185]: ah, I see, we didn't really call AdwApplication

File: com.usebottles.bottles.dev.yml (hunk 4 above)
Comment (mirkobrombin) [reply com.usebottles.bottles.dev.yml]: Ok

**Comment by mirkobrombin** [comment]
Updated checklist:

Checklist:
- [ ] **python**
  - [ ] **misc**
    - [ ] remove all gtk_dialog.run's

  - [ ] **dialogs**
    - [x] crash
    - [x] dlloverrides
    - [x] drives
    - [x] duplicate
    - [x] envvars
    - [x] gamescope
    - [x] generic
    - [ ] generic_cli (need test)
    - [x] journal
    - [ ] onboard
    - [x] rename
    - [x] runargs

  - [ ] **views**
    - [x] bottle_dependencies
    - [x] bottle_details
    - [x] bottle_installers
    - [ ] bottle_preferences (need testing)
    - [x] bottle_programs
    - [x] bottle_taskmanager
    - [x] bottle_versioning
    - [x] details
    - [ ] importer
    - [x] library
    - [x] list
    - [x] loading
    - [x] new
    - [x] preferences

  - [ ] **widgets**
    - [x] component
    - [x] dependency
    - [x] executable
    - [ ] importer
    - [x] installer
    - [x] library
    - [x] page
    - [x] program
    - [ ] state

  - [ ] cli.in (need test)
  - [x] main
  - [x] operation
  - [ ] operation_cli (need testing)
  - [x] window

- [ ] **ui**
  - [x] about
  - [x] component-entry
  - [ ] dependency-entry (need improvements, installers and deps looks different)
  - [x] details
  - [x] details-bottle
  - [x] details-dependencies
  - [x] details-installers
  - [x] details-preferences
  - [x] details-programs
  - [ ] details-versioning (need improvements)
  - [x] dialog-crash-report
  - [x] dialog-dll-overrides
  - [x] dialog-drives
  - [x] dialog-duplicate
  - [x] dialog-env-vars
  - [x] dialog-gamescope
  - [x] dialog-journal
  - [x] dialog-launch-options
  - [x] dialog-rename
  - [x] dialog-run-args
  - [x] dll-override-entry
  - [x] drive-entry
  - [x] env-var-entry
  - [ ] importer
  - [ ] importer-entry
  - [ ] installer-entry (need improvements, installers and deps looks different)
  - [x] library
  - [x] library-entry
  - [x] list
  - [x] list-entry
  - [x] loading
  - [x] new
  - [ ] onboard
  - [x] preferences
  - [x] program-entry
  - [ ] state-entry
  - [ ] task-entry
  - [x] window

- [ ] **Not blocking**
  - [ ] port ComboBox'es to ComboRow 
  - [ ] use Toasts where applicable
  - [ ] use menu models

- [x] **deprecations**
  - [x] message.py
  - [x] dialog.ui
  - [x] empty.ui
  - [x] message-entry.ui
  - [x] environment-variables.ui
  - [x] elementary.css
  - [x] yaru.css

**Pull request merged**